### PR TITLE
EES-4567 replace react-modal with radix-ui dialog

### PR DESCRIPTION
**Pull request by amyb-hiveit**
Replaces react-modal with [Radix UI Dialog](https://www.radix-ui.com/primitives/docs/components/dialog). This solves some scroll locking issues and removes a blocker for upgrading to React 18 (as react-modal doesn't support it currently).

The implementation of `Modal` and `ModalConfirmation` has changed so that you no longer have to manage the open state of the modal, instead you can provide a `triggerButton` to the modal which will open it, e.g.:
```
<Modal title='My modal' triggerButton={<Button>Click me</Button>}>
<p>The content</p>
</Modal>
```
This works well in the majority of places, but you can also still control the modal via the `open` prop if that is more appropriate.

### CODE REVIEW

File: src/explore-education-statistics-admin/src/pages/release/pre-release/components/PreReleaseInvitePlanModal.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ interface Props {
   onExit: () => void;
 }
 
-const PreReleaseInvitePlanModel = ({
+const PreReleaseInvitePlanModal = ({
```
Comment (ntsim) [inline src/explore-education-statistics-admin/src/pages/release/pre-release/components/PreReleaseInvitePlanModal.tsx:15]: Could `export default function` this

File: src/explore-education-statistics-common/src/components/ModalConfirm.tsx [hunk 2]
```diff
@@ -1,41 +1,52 @@
...
+  onExit,
 }: Props) => {
   const isMounted = useMountedRef();
   const [isDisabled, toggleDisabled] = useToggle(false);
+  const [open, toggleOpen] = useToggle(initialOpen);
 
-  const handleAction = (callback: () => void) => async () => {
+  useEffect(() => {
+    toggleOpen(initialOpen);
+  }, [initialOpen, isMounted, toggleOpen]);
```
Comment (ntsim) [inline src/explore-education-statistics-common/src/components/ModalConfirm.tsx:43]: Shouldn't need `isMounted` here. It's also a ref, so changing it doesn't trigger a re-render anyway.

File: src/explore-education-statistics-frontend/src/modules/find-statistics/components/MobileFilters.tsx [hunk 3]
```diff
@@ -24,24 +23,36 @@ const MobileFilters = ({
   themes,
   totalResults,
   onChange,
-  onClose,
 }: Props) => {
+  const mobileFilterButtonRef = useRef<HTMLButtonElement>(null);
```
Comment (ntsim) [inline src/explore-education-statistics-frontend/src/modules/find-statistics/components/MobileFilters.tsx:27]: What's this for?

File: src/explore-education-statistics-frontend/src/modules/find-statistics/components/Filters.tsx [hunk 4]
```diff
@@ -17,45 +16,25 @@ export type FilterChangeHandler = ({
...
-const Filters = ({
-  releaseType,
-  showMobileFilters = false,
-  themeId,
-  themes,
-  totalResults,
-  onChange,
-  onCloseMobileFilters,
-}: Props) => {
+const Filters = ({ releaseType, themeId, themes, onChange }: Props) => {
```
Comment (ntsim) [inline src/explore-education-statistics-frontend/src/modules/find-statistics/components/Filters.tsx:24]: Perhaps it'd be good to be more specific and call these `FiltersDesktop`?

Would suggest also renaming `MobilesFilters` to `FiltersMobile` to match this so they end up next to each other in the project files.

File: src/explore-education-statistics-frontend/src/modules/find-statistics/components/Filters.tsx [hunk 5]
```diff
@@ -17,45 +16,25 @@ export type FilterChangeHandler = ({
...
-          themes={themes}
-          totalResults={totalResults}
-          onChange={onChange}
-          onClose={onCloseMobileFilters}
-        />
-      )}
-    </>
-  );
+
+  if (!isMobileMedia) {
```
Comment (ntsim) [inline src/explore-education-statistics-frontend/src/modules/find-statistics/components/Filters.tsx:27]: Given that the rendering of `FiltersMobile` / `MobileFilters` is controlled by the media query hook in `FindStatisticsPage`, would suggest we also do this for this component as well to be consistent i.e.

```tsx
{isMobileMedia && (
  <FiltersDesktop ... />
)}
```

Also means we have one less media query listener.

File: src/explore-education-statistics-common/src/components/Modal.tsx [hunk 6]
```diff
@@ -1,95 +1,117 @@
...
   onExit,
+  onOpen,
+  onToggleOpen,
 }: ModalProps) => {
-  const appElement =
-    typeof document !== 'undefined'
-      ? (document.getElementById(process.env.APP_ROOT_ID) as HTMLElement)
-      : undefined;
+  const dialogRef = useRef<HTMLDivElement>(null);
+  const [open, setOpen] = useToggle(initialOpen);
```
Comment (ntsim) [inline src/explore-education-statistics-common/src/components/Modal.tsx:46]: Could be `toggleOpen` instead of `setOpen`.

**Review by ntsim — CHANGES_REQUESTED** [review]

File: src/explore-education-statistics-admin/src/pages/admin-dashboard/components/DraftReleasesTable.tsx [hunk 7]
```diff
@@ -98,18 +83,12 @@ const DraftReleasesTable = ({
                       Publication / Release period
                     </th>
                     <th>
-                      Status{' '}
-                      <ButtonText onClick={toggleDraftStatusGuidance.on}>
-                        <InfoIcon description="Guidance on draft states" />
-                      </ButtonText>
+                      Status <DraftStatusGuidanceModal />
```
[the remark this answers is not in the document]
Comment (amyb-hiveit) [reply src/explore-education-statistics-admin/src/pages/admin-dashboard/components/DraftReleasesTable.tsx:90]: Yep!

File: src/explore-education-statistics-admin/src/pages/release/components/__tests__/ReleaseStatusForm.test.tsx [hunk 8]
```diff
@@ -816,7 +816,7 @@ describe('ReleaseStatusForm', () => {
         />,
       );
 
-      await userEvent.type(
+      userEvent.type(
```
[the remark this answers is not in the document]
Comment (amyb-hiveit) [reply src/explore-education-statistics-admin/src/pages/release/components/__tests__/ReleaseStatusForm.test.tsx]: Thanks for the heads up. I've changed this bach so it'll be one less to change when we do the update.

File: src/explore-education-statistics-frontend/src/modules/find-statistics/components/MobileFilters.tsx (hunk 3 above)
Comment (amyb-hiveit) [reply src/explore-education-statistics-frontend/src/modules/find-statistics/components/MobileFilters.tsx]: It was left over from the previous version of the modal, it was used to refocus the button when the modal was closed but it's not needed anymore so I've removed it.

**Review by ntsim — APPROVED** [review]

**Pull request merged**